src/main/scala/se/nimsa/dcm4che/streams/DicomParts.scala

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2017 Lars Edenbrandt
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package se.nimsa.dcm4che.streams
+
+import akka.util.ByteString
+import org.dcm4che3.data.{SpecificCharacterSet, VR}
+import org.dcm4che3.util.ByteUtils
+
+
+object DicomParts {
+
+  trait DicomPart {
+    def bigEndian: Boolean
+
+    def bytes: ByteString
+  }
+
+  case class DicomPreamble(bytes: ByteString) extends DicomPart {
+    def bigEndian = false
+  }
+
+  case class DicomHeader(tag: Int, vr: VR, length: Int, isFmi: Boolean, bigEndian: Boolean, explicitVR: Boolean, bytes: ByteString) extends DicomPart {
+
+    def updateLength(newLength: Int) : DicomHeader = {
```

Comment:
This may be very subjective but this name (and similar for other case classes here) makes it sound like the object is mutated. I would prefer if it was called something like `withUpdatedLength`, or moved to a companion object with the signature `updateLength(dicomHeader: DicomHeader, newLength: Int)`